src/dao/DAO.js

```diff
@@ -188,6 +189,45 @@ class DAO<TEntity: { id: EntityID }, TEntityMutator> extends BaseDAO<
     );
   }
 
+  // todo fix flow inconsistensy TResult with entityLoaders - TEntity
+  fetchCustom<TResult>(customQuery: string): LoadObject<TResult> {
```

Comment:
If you support `apply` then you can just add that to the `QueryOptions`.

We will still need `fetchCustom` since `fetchByIDs` won't work in this case.

---

It would also be good to add another cache that looks like `Map<string, Object>` that can hold these custom queries instead of holding it in the main cache.